### DISCUSSION TITLE
Why does a question posed to copilot in vscode have more detail than the same question posed in copilot4eclipse?

### DISCUSSION THREAD

**Discussion by davidmichaelkarr** [Genuitec/Copilot4Eclipse-community#11] [Help]
I have Github Copilot configured in vscode, and also in Eclipse with copilot4eclipse. I have the newest versions of all of these.

If I ask the same question in the chat window of both tools, I get mostly the same information, but I'm finding that the response from vscode has more detail, and is formatted differently, the vscode output generally looks a little better.  Why exactly would that be?

Also, on a pure functional point, I notice that in vscode, after entering the question, it prints something like "working ...", and then it erases that when it prints the response. In copilot4eclipse, once you enter the question, nothing happens for several seconds. It makes you think it isn't doing anything.

**Comment by wayneparrott** [comment]
Hi David,

Thanks for the constructive feedback.

>  the vscode output generally looks a little better

Can you provide a screenshot or two for consideration?

As for the quality of content, context is everything. This includes other files you have opened in the IDE, previous prompts, etc. I'll do some comparisons soon with 

> I notice that in vscode, after entering the question, it prints something like "working ...

We recognize the importance of a responsive UX and are working on providing snappier feedback similar to what you describe. 

W

**Comment by davidmichaelkarr** [comment]
These two are at different zoom levels, so zoom in to see text differences.

This is what I see in vscode:

![image](https://github.com/user-attachments/assets/e06c0c21-7178-4265-b822-42a6bc79d5d7)

This is what I see in copilot4eclipse:

![image](https://github.com/user-attachments/assets/49b1d55e-30f9-4247-9ce7-26762c412b89)

I notice in particular the information in Python's concurrency model info that is present in the vscode sample, but not in the copilot4eclipse output.

**Comment by miller79** [comment]
I just saw this question and am curious myself.  I noticed in VSCode that it is using the model GPT-4-Turbo where this plugin is only using GPT-4. Is there a way to modify which model is being used behind the scenes? I've noticed in PyCharm that it uses GPT-4o.

**Reply by wayneparrott** [reply]
More model options are on the way but VSCode will get the goodies before other IDEs: https://github.blog/news-insights/product-news/bringing-developer-choice-to-copilot/

To your question why does VSCode get all the good stuff before the other IDES? Simple, it's a business decision for Microsoft and GitHub. VSCode has the largest developer adoption. So it makes sense to deliver Copilot features to the largest audience first. Then propagate those features to lesser IDEs shortly thereafter.

Personally, I'm looking forward to being able to use Claude through Copilot4Eclipse in Eclipse. I do not have a timeframe for when Copilot4Eclipse will have this support but it will be asap.

Wayne
Copilot4Eclipse Team

**Reply by miller79** [reply]
Interesting! Makes sense but doesn't seem completely fair but it is their product to do what the see fit. Thanks for the info!
